### ISSUE TITLE
Don't use pointers for interface{} types

### ISSUE THREAD

**Issue by agorman** [Nerzal/gocloak#187]
**Describe the bug**
In the recent refactor of all types to points you have two places where you use interfaces as pointers.

https://github.com/Nerzal/gocloak/blob/master/models.go#L605
https://github.com/Nerzal/gocloak/blob/master/models.go#L662

Because of this I can no longer set those values. It's possible there is an easy way to do it but I haven't figured it out. If so let me know and perhaps this isn't actually a bug.

When trying to set `RealmRepresentation.Roles` I get an error like:

~~~
./organization.go:149:3: cannot use roles (type *map[string]interface {}) as type *interface {} in field value:
	*interface {} is pointer to interface, not interface
~~~

I believe reverting back to `interface{}` will solve the issue. If the value isn't set then the JSON representation would be `null`.

**To Reproduce**
Set `RealmRepresentation.Roles` to any pointer value.

**Expected behavior**
I can set `RealmRepresentation.Roles`.

**Comment by SVilgelm** [comment]
Oh, sorry, my bad

**Comment by agorman** [comment]
Thanks, you rock!

**Comment by SVilgelm** [comment]
@agorman I added `RolesRepresentation` struct for the `Roles` field, but I don't understand how to create Keycloak's MultiHashMap for the `Components` field

**Comment by agorman** [comment]
I actually only use the `Roles` field so I'm not sure either. A `RolesRepresentation` struct is appreciated though.

**Comment by SVilgelm** [comment]
Could you please test it? You can install using `go get github.com/Nerzal/gocloak/v6@e6dddbe826bb33073c29bbe8220d1fed6766ed4e`

**Comment by agorman** [comment]
Yes, but I won't be able to get to it until tomorrow.

**Comment by SVilgelm** [comment]
np, I think @Nerzal will see this tomorrow as well

**Comment by SVilgelm** [comment]
I mean need to wait anyway

**Comment by agorman** [comment]
I took a quick look at the code and I think client will need to be a `map[string]interface{}` or I think it could be a `map[string][]*Role` if a `Composites` field was added to the `Role` struct.

Here is what I'm passing it currently.

~~~
roles := map[string]interface{}{
	"realm": []*gocloak.Role{
		{
			Name: gocloak.StringP("owner"),
		},
	},
	"client": map[string]interface{}{
		"channels": []map[string]interface{}{
			{
				"name": "viewer",
			},
			{
				"name":      "user",
				"composite": true,
				"composites": map[string]interface{}{
					"client": map[string]interface{}{
						"channels": []string{
							"viewer",
						},
					},
				},
			},
			{
				"name":      "admin",
				"composite": true,
				"composites": map[string]interface{}{
					"client": map[string]interface{}{
						"channels": []string{
							"viewer",
							"user",
						},
					},
				},
			},
		},
	},
}
~~~

**Comment by SVilgelm** [comment]
Ok, thank you, I change the `Client`'s type to `*map[string][]Role`

**Comment by agorman** [comment]
Can you please add the following

~~~
// CompositesRepresentation represents the composite roles of a role
type CompositesRepresentation struct {
	Client *map[string][]string `json:"client,omitempty"`
	Realm  *[]string            `json:"realm,omitempty"`
}
~~~

~~~
// Role is a role
type Role struct {
	ID                 *string                   `json:"id,omitempty"`
	Name               *string                   `json:"name,omitempty"`
	ScopeParamRequired *bool                     `json:"scopeParamRequired"`
	Composite          *bool                     `json:"composite"`
	Composites         *CompositesRepresentation `json:"composites"`
	ClientRole         *bool                     `json:"clientRole"`
	ContainerID        *string                   `json:"containerId,omitempty"`
	Description        *string                   `json:"description,omitempty"`
	Attributes         *map[string][]string      `json:"attributes,omitempty"`
}
~~~


That way I can create the composites like this:

~~~
	roles := &gocloak.RolesRepresentation{
		Realm: &[]gocloak.Role{
			{
				Name: gocloak.StringP("owner"),
			},
		},
		Client: &map[string][]gocloak.Role{
			"channels": []gocloak.Role{
				{
					Name: gocloak.StringP("viewer"),
				},
				{
					Name:      gocloak.StringP("user"),
					Composite: gocloak.BoolP(true),
					Composites: &gocloak.CompositesRepresentation{
						Client: &map[string][]string{
							"channels": []string{
								"viewer",
							},
						},
					},
				},
				{
					Name:      gocloak.StringP("admin"),
					Composite: gocloak.BoolP(true),
					Composites: &gocloak.CompositesRepresentation{
						Client: &map[string][]string{
							"channels": []string{
								"viewer",
								"user",
							},
						},
					},
				},
			},
		},
	}
~~~

**Comment by SVilgelm** [comment]
released in v6.3.1
